### ISSUE TITLE
Allow `group_by_dynamic` to operate on data of type `time`

### ISSUE THREAD

**Issue by thomascamminady** [pola-rs/polars#13560]
### Description

This relates to [a question I posted on Stack Overflow](https://stackoverflow.com/questions/77785396/run-group-by-dynamic-in-polars-but-only-on-timestamp/77786272?noredirect=1#comment137130649_77786272).

I would like to call `group_by_dynamic`, but omit the date information and only group over the timestamps. The proposed solution in the SO question suggests to create a new date column that contains the original hours/minutes/seconds, but adds a fake date that is identical for all rows. Then apply `group_by_dynamic`, and then basically discard the date information again.

Here's the proposed solution
```python
df.with_columns(time=pl.date(2024, 1, 1).dt.combine(pl.col("datetime").dt.time())).sort(
    "time"
).group_by_dynamic("time", every="10m", period="1h").agg(
    pl.col("duration_in_traffic_s").mean()
).with_columns(
    time=pl.col("time").dt.time()
)
```

Is there a fundamental reason why `group_by_dynamic` can't work with **only** the time data and not just with datetime data?

**Comment by thomascamminady** [comment]
Just to be more precise, I wonder whether this should be possible

```python
df.with_columns(
    time=pl.col("time").dt.time()
).group_by_dynamic("time", every="10m", period="1h").agg(
    pl.col("duration_in_traffic_s").mean()
)
```
which currently results in this error:
```
ComputeError: expected any of the following dtypes: { Date, Datetime, Int32, Int64 }, got time
```

**Comment by auxym** [comment]
Agreed, and I believe it should also support type `Duration`, which as I understand is meant to represent "some duration of time since an arbitrary/unknown epoch".

**Comment by flokoenig** [comment]
First of all, polars is a really great step up from pandas. But I, too, would very much like to see relative times as "first class citizens".

I think it natural to run a group_by_dynamic on a duration column, but it doesn't work. And as someone with a science background, I cannot for the life of me understand why frameworks like polars (and pandas) seem to assume that time series have to be based on absolute dates/times. 99% of the time series I have ever had to work with start at t=0 and are relative to that -- and just accepting it and treating t=0 as 00:00:00 UTC on 1 January 1970 is a really awkward, unnatural and unnecessary hack.
